Source/qol/floatinginfobox.cpp

```diff
@@ -0,0 +1,480 @@
+			reqStr = UiFlags::ColorRed;
+		if (item._iMinDex > myPlayer._pDexterity)
+			reqDex = UiFlags::ColorRed;
+		if (item._iMinMag > myPlayer._pMagic)
+			reqMag = UiFlags::ColorRed;
+
+		// Add the lines vector contents into the linesWithColor vector, which contains both the strings and the colors that they will be
+		std::vector<DrawStringFormatArg> linesWithColor;
+		UiFlags color = UiFlags::ColorWhite;
+		for (size_t i = 0; i < lines.size(); i++) {
+			const auto &line = lines[i];
+			// Item name and base item name should match the item quality color, normally found in InfoColor
+			// Note: Split lines are both considered separate lines, even though in game they appear on the same line visually
+			if (line.find(item._iIName) != std::string::npos || line.find(item._iName) != std::string::npos) {
+				color = item.getTextColor();
+			} else if (line.find("Armor:") != std::string::npos || line.find("Damage:") != std::string::npos || line.find("Durability:") != std::string::npos) {
+				color = UiFlags::ColorWhite;
+			} else if ((i > 0 && lines[i - 1].find("Armor:") != std::string::npos)) {
+				color = armorBonus;
+			} else if ((i > 0 && lines[i - 1].find("Damage:") != std::string::npos)) {
+				color = damageBonus;
+			} else if ((i > 0 && lines[i - 1].find("Durability:") != std::string::npos)) {
+				color = durBonus;
+			} else if ((line.find("Required Strength:") != std::string::npos) || (i > 0 && lines[i - 1].find("Required Strength:") != std::string::npos)) {
+				color = reqStr;
+			} else if ((line.find("Required Dexterity:") != std::string::npos) || (i > 0 && lines[i - 1].find("Required Dexterity:") != std::string::npos)) {
+				color = reqDex;
+			} else if ((line.find("Required Magic:") != std::string::npos) || (i > 0 && lines[i - 1].find("Required Magic:") != std::string::npos)) {
+				color = reqMag;
+			} else if (line.find("Not Identified") != std::string::npos) {
+				color = UiFlags::ColorWhite;
+			} else if (item._iMagical != ITEM_QUALITY_NORMAL) {
+				// All remaining lines for non-normal items will be powers, so set all to blue
+				color = UiFlags::ColorBlue;
+			}
+
+			linesWithColor.emplace_back(line, color);
+		}
+
+		// linesBase is used to place the {} and newlines to grab the actual string data
+		// Lines that contain a colon may have a different color for the second part, so two {}'s are added
+		std::string linesBase;
+		bool previousLineHadColon = false;
+		for (size_t i = 0; i < lines.size(); i++) {
+			bool currentLineHasColon = (lines[i].find(":") != std::string::npos);
+
+			if (!previousLineHadColon && (!currentLineHasColon || (i > 0 && currentLineHasColon))) {
+				if (currentLineHasColon) {
+					linesBase.append("{}{}");
+				} else {
+					linesBase.append("{}");
+				}
+				if (i != lines.size() - 1) {
+					linesBase.append("\n");
+				}
+			}
+
+			previousLineHadColon = currentLineHasColon;
+		}
+
+		std::string_view linesBaseView(linesBase.data(), linesBase.length());

```

Comment:
Use the polyfill until 1.6.0-dev
```suggestion
		string_view linesBaseView(linesBase.data(), linesBase.length());
```